### PR TITLE
added GeometryGroup

### PR DESCRIPTION
**Pull request by tylerflex**
To do
- [x] Add tests
- [x] Integrate with backend
- [x] Implement a custom `.inside()`

### CODE REVIEW

File: tidy3d/components/geometry.py [hunk 1]
```diff
@@ -2059,3 +2060,81 @@ def normalize(v):
...
+    geometries: List[GeometryType] = pydantic.Field(
+        [],
+        title="Geometries",
+        description="List of geometries in a single grouping. "
+        "Can provide significant performance enhancement in ``Structure`` when all geometries are "
+        "assigned the same medium.",
+        min_items=1,
+    )
+
+    @property
```
Comment (momchil-flex) [inline tidy3d/components/geometry.py:2077]: Should we cache this?

Comment (tylerflex) [reply tidy3d/components/geometry.py:2077]: Geometry doesn't have a `_hash`, otherwise we could just cache all properties.  There was this issue with multiple inheritance that never got resolved.

Comment (tylerflex) [reply tidy3d/components/geometry.py:2077]: right now, only `Simulation` has a hash

Comment (momchil-flex) [reply tidy3d/components/geometry.py:2077]: Every model has a `__hash__` thought. Are you just saying that `make_static` will not work?

Comment (momchil-flex) [reply tidy3d/components/geometry.py:2077]: As in it will not make the `GeometryGroup` static. I think that could still be fine, the hash may be faster than getting all the geometry bounds?

Comment (tylerflex) [reply tidy3d/components/geometry.py:2077]: hm, we could give it a try. (move `cache` to `base.py` I guess?)

File: tidy3d/components/simulation.py [hunk 2]
```diff
@@ -1592,14 +1569,9 @@ def get_eps(medium: Medium, freq: float):
...
             """returns epsilon data on grid of points defined by coords"""
             xs, ys, zs = coords.x, coords.y, coords.z
-            rmin = tuple(coord[0] for coord in (xs, ys, zs))
-            rmax = tuple(coord[-1] for coord in (xs, ys, zs))
-            points_box = Box.from_bounds(rmin=rmin, rmax=rmax)
             x, y, z = np.meshgrid(xs, ys, zs, indexing="ij")
             eps_array = eps_background * np.ones(x.shape, dtype=complex)
             for structure in self.structures:
-                if not points_box.intersects(structure.geometry):
-                    continue
```
Comment (momchil-flex) [inline tidy3d/components/simulation.py:1602]: Why was this removed? Is it some weird mix up in creating the geo_group PR before this was added? But I'm confused because even after rebasing against develop, it is still missing (so if we merge this PR it will remove it).

Comment (tylerflex) [reply tidy3d/components/simulation.py:1602]: I think this was a mistake, maybe it's because I made this branch before we started skipping non-intersecting boxes?

Comment (tylerflex) [reply tidy3d/components/simulation.py:1602]: we can apply this commit to the PR?
https://github.com/flexcompute/tidy3d/commit/9f7cf740119720a82fee1393e90b4611b3e71d23

Comment (momchil-flex) [reply tidy3d/components/simulation.py:1602]: Yeah ok I was just checking that there wasn't anything intentional here. I'll fix it.
